ReactNativeClient/lib/services/ExternalEditWatcher.js

```diff
@@ -211,12 +195,52 @@ class ExternalEditWatcher {
 					clearInterval(iid);
 					reject(wrapError(error));
 				});
+				subProcess.on('close', onClose);
 			} catch (error) {
 				throw wrapError(error);
 			}
 		});
 	}
 
+	async spawnDefault(file, onClose) {
+		// windows
+		let path = '';
+
+		if (shim.isWindows()) {
+			path = 'start';
+		} else if (shim.isMac()) {
+			path = 'open';
+		} else if (shim.isLinux() || shim.isFreeBSD()) {
+			path = 'xdg-open';
+		} else {
+			// Fallback on the electron open method
+			bridge().openExternal(`file://${file}`);
+		}
+
+		return this.spawn(path, [file], {}, onClose);
+	}
+
+	async spawnCommand(path, args, options, onClose) {
```

Comment:
An unwritten coding convention is that if there's an `options` argument it should always be last. So I think in your case, you could simply make `onClose` to be one of the options. Same for the `spawn` method.

Reply:
Good call, I thought this was kinda ugly.

Reply:
Changed in the latest commit